File: EssentialToolsCs/EssentialToolsCs/Controllers/HomeController.cs
namespace EssentialToolsCs.Controllers
{
    using System.Web.Mvc;
    using Models;
    using Ninject;

    // using Ninject;

    public class HomeController : Controller
    {
        private readonly IValueCalculator calc;
        private Product[] products =
        {
            new Product { Name = "Kayak", Category = "Watersports", Price = 275M }, 
            new Product { Name = "Lifejacket", Category = "Watersports", Price = 48.95M }, 
            new Product { Name = "Soccer ball", Category = "Soccer", Price = 19.50M }, 
            new Product { Name = "Corner flag", Category = "Soccer", Price = 34.95M }
        };

        public HomeController(IValueCalculator calcParam)
        {
            calc = calcParam;
        }

        // GET: Home
        public ActionResult Index()
        {
            //var ninjectKernel = new StandardKernel();
            //ninjectKernel.Bind<IValueCalculator>().To<LinqValueCalculator>();

            // var calc = new LinqValueCalculator();
            // var calc = ninjectKernel.Get<IValueCalculator>();
            var cart = new ShoppingCart(calc) {Products = products};
            var totalValue = cart.CalculateProductTotal();
            return View(totalValue);
        }
    }
}

File: EssentialToolsVb0001/EssentialsToolsVb/Controllers/HomeController.vb
Imports System.Web.Mvc
Imports Ninject

Namespace Controllers
    Public Class HomeController
        Inherits Controller

        Dim calc As IValueCalculator
        Dim products = New List(Of Product) From
            {
            New Product With {.Name = "Kayak", .Category = "Watersports", .Price = 275D},
            New Product With {.Name = "Lifejacket", .Category = "Watersports", .Price = 48.95D},
            New Product With {.Name = "Soccer ball", .Category = "Soccer", .Price = 19.5D},
            New Product With {.Name = "Corner flag", .Category = "Soccer", .Price = 34.95D}
        }

        Public Sub New(calcParam As IValueCalculator)
            calc = calcParam
        End Sub

        ' GET: Home
        Function Index() As ActionResult
            Dim ninjectKernel As New StandardKernel
            ninjectKernel.Bind(Of IValueCalculator).To(Of LinqValueCalculator)()

            ' Dim calc As New LinqValueCalculator
            Dim calc = ninjectKernel.Get(Of IValueCalculator)
            Dim cart As New ShoppingCart(calc) With {.Products = products}
            Dim totalValue = cart.CalculateProductTotal()
            Return View(totalValue)
        End Function
    End Class
End Namespace